locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsSshMachineLocation.java

```diff
@@ -162,8 +162,12 @@ public void init() {
     @Override
     public void rebind() {
         super.rebind();
-        ComputeServiceContext context = jcloudsParent.getComputeService().getContext();
-        runScriptFactory = context.utils().injector().getInstance(RunScriptOnNode.Factory.class);
+        
+        if (jcloudsParent != null) {
+            // can be null on rebind, if location has been "orphaned" somehow
+            ComputeServiceContext context = jcloudsParent.getComputeService().getContext();
+            runScriptFactory = context.utils().injector().getInstance(RunScriptOnNode.Factory.class);
+        }
```

Comment:
Isn't the rebind failing a good thing in this case - it shows that something is wrong and there will be runtime errors if not addressed.
Can we address the problem in another way? For example:
  * why are parent locations not serialized - is this a bug?
  * serialize parent locations even if not managd